src/util/ConfigManager/ConfigManager.cpp

```diff
@@ -277,10 +293,18 @@ void ConfigManager::parseConfig(const nlohmann::json& j) {
   const auto& jFlattend = j.flatten();
 
   // All the configuration options together with their paths.
-  std::vector<std::pair<std::string, ConfigOption*>> allConfigOption =
-      configurationOptions("");
-  ad_utility::HashMap<std::string, ConfigOption*> allConfigOptionHashMap(
-      allConfigOption.begin(), allConfigOption.end());
+  const auto allConfigOptions{[this]() {
+    std::vector<std::pair<std::string, ConfigOption&>> allConfigOption =
+        configurationOptions();
+
+    // You can't put references into hash maps. Instead, we use pointer.
+    auto pointerVersion = std::views::transform(
+        allConfigOption, [](const std::pair<std::string, ConfigOption&>& p) {
+          return std::make_pair(p.first, &p.second);
+        });
```

Comment:
It works, if you
a) Use `std::unordered_map` (only doesn't work for the absl containers)
b) Don't use the `[]` operator (you have to use the `at()` function.
I would prefer this (with the same comment why we use `unordered_map` isntead of our default `HashMap`, but then the rest of the code gets simpler again.

Reply:
Done, see [bd0aeb9](https://github.com/ad-freiburg/qlever/pull/1071/commits/bd0aeb9a0809e19991996972bd0505673b1b7c8a).